### PR TITLE
js3-week3-homework

### PR DESCRIPTION
**Pull request by amerdahabi**

### CODE REVIEW

File: javascript/javascript3/week3/canvas/main.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  }
+}
+
+const c1 = new Circle(50, 150, 20, 0, 2 * Math.PI, "#000000");
+//c1.draw();
+
+// Now lets make art!
+
+setInterval(() => {
+  const randomX = Math.floor(Math.random() * canvas.width);
```
Comment (aleksd-dfds) [inline javascript/javascript3/week3/canvas/main.js]: a function GetRandomInteger(max) would save you some typing and simplify the code

File: javascript/javascript3/week3/canvas/main.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+let mouseX = 0;
+let mouseY = 0;
+
+canvas.addEventListener("mousemove", mousePosition);
+
+function mousePosition(e) {
+  mouseX = e.clientX - ctx.canvas.offsetLeft;
+  mouseY = e.clientY - ctx.canvas.offsetTop;
+  const randomR = Math.floor(Math.random() * 20);
+  const randomColor = "#" + Math.floor(Math.random() * 16777215).toString(16);
```
Comment (aleksd-dfds) [inline javascript/javascript3/week3/canvas/main.js]: This should be a function called getRandomColor, and the number 16777215 will confuse the reader (what is that anyway?)

File: javascript/javascript3/week3/gittingIntoPromises/main.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
+console.log("Gitting Into Promises");
+
+const container = document.querySelector(".container");
+const userName = '';
+
+function fetchData(userName) {
```
Comment (aleksd-dfds) [inline javascript/javascript3/week3/gittingIntoPromises/main.js:6]: fetchRepository is probably a better name

File: javascript/javascript3/week3/shoppingCart/main.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  addProduct(product) {
+    // Implement functionality here
+    this.products.push(product);
+  }
+
+  removeProduct(product) {
+    // Implement functionality here
+    if (!product) {
+      return;
+    } else {
```
Comment (aleksd-dfds) [inline javascript/javascript3/week3/shoppingCart/main.js]: Can be rewritten shorter as:
```
if (!product) {
    return;
}
this.products = this.products.filter(pro => pro.name !== product.name);
```

File: javascript/javascript3/week3/shoppingCart/main.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+      this.products = this.products.filter(pro => pro.name !== product.name);
+    
+    }
+  }
+
+  searchProduct(productName) {
+    // Implement functionality here
+    if (!productName) {
+      return;
+    } else {
```
Comment (aleksd-dfds) [inline javascript/javascript3/week3/shoppingCart/main.js]: same here: "else" is not necessary

File: javascript/javascript3/week3/shoppingCart/main.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    // Implement functionality here
+    this.totalPrice = this.products.reduce((a, b) => a + b.price, 0);
+  }
+
+  renderProducts() {
+    // Implement functionality here
+    // i'm working on it 
+  }
+
+  getUser() {
```
Comment (aleksd-dfds) [inline javascript/javascript3/week3/shoppingCart/main.js:48]: good

File: javascript/javascript3/week3/gittingIntoPromises/main.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+function promiseAll(){
+    Promise.all([
+        fetchData('amerdahabi'),
+        fetchData('SwathiSankararaman'),
+        fetchData("benhamid")
+    ])
+}
+promiseAll();
```
Comment (aleksd-dfds) [inline javascript/javascript3/week3/gittingIntoPromises/main.js:45]: Promise.all, should be followed by .then(). Otherwise it's doing nothing. Fix this.

**Review by aleksd-dfds — COMMENTED** [review]
Good job, some small comments, but also please fix your Promise.all to do something useful.

File: javascript/javascript3/week3/shoppingCart/main.js (hunk 6 above)
Comment (amerdahabi) [reply javascript/javascript3/week3/shoppingCart/main.js]: thank you so much Aleks for the feedback and all the notes that you mentioned.
the cod looks better now (;

**Comment by amerdahabi** [comment]
thank you so much Aleks for the feedback and all the notes that you mentioned.
the cod looks better now (;

**Pull request merged**